dead/differential_testing.py

```diff
@@ -51,24 +51,33 @@ class DifferentialTestingCase:
     markers_only_eliminated_by_setting1: tuple[DCEMarker | VRMarker, ...]
     markers_only_eliminated_by_setting2: tuple[DCEMarker | VRMarker, ...]
 
+    def __post_init__(self) -> None:
+        assert set(self.markers_only_eliminated_by_setting1).isdisjoint(
+            self.markers_only_eliminated_by_setting2
+        )
+
 
 class DifferentialTestingMode(Enum):
     """
-    - Unidirectional: a marker is interesting only if the first
-      compilation setting missed it and the second eliminated
-    - Bidirectional: a marker is interesting if any of the compilation settings
-      missed it and the other found it
+    - Unidirectional: any marker is interesting only if the first
+      compilation setting missed it and the second eliminated it
+    - Bidirectional: any marker is interesting if any of the compilation
+      settings missed it and the other eliminated it
+    - MarkerMissedByFirst: a particular marker is interesting if the
+      first compilation setting missed it and the other eliminated it
     """
 
-    Unidirectional = 0
+    Unidirectional = 0  # AnyMissedByFirst?
     Bidirectional = 1
+    MarkerMissedByFirst = 2
 
 
 def differential_test(
     program: SourceProgram,
     setting1: CompilationSetting,
     setting2: CompilationSetting,
     testing_mode: DifferentialTestingMode = DifferentialTestingMode.Bidirectional,
+    missed_marker: DCEMarker | VRMarker | None = None,
```

Comment:
I don't think we can/should mix the test with which we find the cases and the one with which we retest/reduce a case.

Also, with the current implementation you seem to instrument the program two or more times: Once when finding the case and then again every iteration of the reduction callback?

Reply:
Oops, good point. I'll refactor